File: info/1. Алгоритмы/5. Вычислительные алгоритмы/15. Хеш-функция/code sample/Python/sample.py
def text_to_number(text):
    radix = 256
    result = 0
    n = len(text)
    for i in range(n):
        result += ord(text[i]) * radix ** (n - 1 - i)
    return result

class HashService:

    def __init__(self, key_range, collision_level):
        self.prime_number = {10:7,20:19,50:47,100:97,250:241,500:499,1000:997,2500:2477,5000:4999,10000:9973}
        self.m = key_range // collision_level
        for k in self.prime_number.keys():
            if self.m < k:
                self.m = k
                break
    
    def generate_hash(self, text):
        return text_to_number(text) % self.m


text = "Hello world"

hash_service = HashService(2500,4)
print(hash_service.generate_hash(text))

File: info/1. Алгоритмы/5. Вычислительные алгоритмы/15. Хеш-функция/code sample/Fortran/sample.f90
program sample
use hash_service
implicit none
type(Pirson_hash)::pir_hash
character(len = 5)::text

text = 'Hello'

call pir_hash%init()

write(*,*) pir_hash%generate_hash(text)
write(*,'(B32)') pir_hash%generate_hash(text)

write(*,*) pir_hash%generate_four_byte_hash(text)
write(*,'(B32)') pir_hash%generate_four_byte_hash(text)

end program sample
